SourceryTests/Models/TypeNameSpec.swift

```diff
@@ -62,6 +62,23 @@ class TypeNameSpec: QuickSpec {
                 it("reports array correctly") {
                     expect(TypeName("[Int]").isArray).to(beTrue())
                     expect(TypeName("[[Int]]").isArray).to(beTrue())
+                    expect(TypeName("[[Int: Int]]").isArray).to(beTrue())
+                }
+            }
+
+            context("given dictionary type") {
+                context("as name") {
+                    it("reports dictionary correctly") {
+                        expect(TypeName("[Int: Int]").isDictionary).to(beTrue())
+                        expect(TypeName("[[Int]: [Int]]").isDictionary).to(beTrue())
+                        expect(TypeName("[Int: [Int: Int]]").isDictionary).to(beTrue())
```

Comment:
Yes, isArray should be true, and isDictionary should be false. Just add this case here and maybe some more negative cases so that we don't have regressions here.